src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -2368,6 +2370,7 @@ private BoundExpression BindNameofOperatorInternal(InvocationExpressionSyntax no
             CheckFeatureAvailability(node, MessageID.IDS_FeatureNameof, diagnostics);
             var argument = node.ArgumentList.Arguments[0].Expression;
             var boundArgument = BindExpression(argument, diagnostics);
+            boundArgument = ResolveToExtensionMemberIfPossible(boundArgument, diagnostics);
```

Comment:
:memo: `BindToNaturalType` (called below) also does this. I've been debating whether to remove the redundant call. We'll discuss